core/src/main/scala/cats/syntax/option.scala

```diff
@@ -105,6 +105,9 @@ final class OptionOps[A](private val oa: Option[A]) extends AnyVal {
   def toInvalidNel[B](b: => B): ValidatedNel[A, B] =
     oa.fold[ValidatedNel[A, B]](Validated.Valid(b))(Validated.invalidNel)
 
+  def toInvalidA[F[_]: Applicative, B](b: => B): Validated[F[A], B] =
+    oa.fold[Validated[F[A], B]](Validated.Valid(b))(Validated.invalidA[F, A, B])
```

Comment:
I think, this is the most disputable place. On one hand, the `B` type param might seem unavoidable, but on the other hand it would result to a quite inconvenient user experience:
```scala
val maybeResult: Option[MyResult] = ???

maybeResult.toInvalidA[NonEmptyVector, MyError](MyError("failed to get result"))
```
Personally, I would prefer to avoid repeating `MyError`. I think it would be better to use the following approach here as well:
https://github.com/typelevel/cats/blob/8373e21a4f112abe298713f78540be93554ff5d2/core/src/main/scala/cats/syntax/option.scala#L356

Reply:
yep, agree.